core/test/ModelMetadata_test.cpp

```diff
@@ -0,0 +1,149 @@
+        exit(1);
+    }
+
+    // This metadata is specific to the periodic boundary conditions.
+    // They are all zero because the input metadata file `partitionFilenameCB` does not use periodic
+    // boundary conditions.
+    REQUIRE(metadata.neighbourRanksPeriodic[ModelMetadata::LEFT].size() == 0);
+    REQUIRE(metadata.neighbourRanksPeriodic[ModelMetadata::RIGHT].size() == 0);
+    REQUIRE(metadata.neighbourRanksPeriodic[ModelMetadata::BOTTOM].size() == 0);
+    REQUIRE(metadata.neighbourRanksPeriodic[ModelMetadata::TOP].size() == 0);
+}
+
+MPI_TEST_CASE("Test getPartitionMetadata periodic boundary", 3)
+{
+    ModelMetadata metadata(partitionFilenamePB, test_comm);
+    REQUIRE(metadata.mpiComm == test_comm);
+    // this metadata should be identical to the Closed Boundary version so we check it again
+    if (test_rank == 0) {
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::LEFT].size() == 0);
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::RIGHT] == std::vector<int> { 2 });
+        REQUIRE(metadata.neighbourExtents[ModelMetadata::RIGHT] == std::vector<int> { 4 });
+        REQUIRE(metadata.neighbourHaloStarts[ModelMetadata::RIGHT] == std::vector<int> { 0 });
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::BOTTOM].size() == 0);
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::TOP] == std::vector<int> { 1 });
+        REQUIRE(metadata.neighbourExtents[ModelMetadata::TOP] == std::vector<int> { 7 });
+        REQUIRE(metadata.neighbourHaloStarts[ModelMetadata::TOP] == std::vector<int> { 0 });
+    } else if (test_rank == 1) {
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::LEFT].size() == 0);
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::RIGHT] == std::vector<int> { 2 });
+        REQUIRE(metadata.neighbourExtents[ModelMetadata::RIGHT] == std::vector<int> { 5 });
+        REQUIRE(metadata.neighbourHaloStarts[ModelMetadata::RIGHT] == std::vector<int> { 12 });
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::BOTTOM] == std::vector<int> { 0 });
+        REQUIRE(metadata.neighbourExtents[ModelMetadata::BOTTOM] == std::vector<int> { 7 });
+        REQUIRE(metadata.neighbourHaloStarts[ModelMetadata::BOTTOM] == std::vector<int> { 21 });
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::TOP].size() == 0);
+    } else if (test_rank == 2) {
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::LEFT] == std::vector<int> { 0, 1 });
+        REQUIRE(metadata.neighbourExtents[ModelMetadata::LEFT] == std::vector<int> { 4, 5 });
+        REQUIRE(metadata.neighbourHaloStarts[ModelMetadata::LEFT] == std::vector<int> { 6, 6 });
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::RIGHT].size() == 0);
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::BOTTOM].size() == 0);
+        REQUIRE(metadata.neighbourRanks[ModelMetadata::TOP].size() == 0);
+    } else {
+        std::cerr << "only valid for 3 ranks" << std::endl;
+        exit(1);
+    }
```

Comment:
Could this common part of the tests be pulled out into a function that's called in both places? It'd avoid duplication and reduce the likelihood of human errors when copy/pasting and reviewing.